### PR TITLE
Added snackbar notifications for all of the professor's actions

### PR DESCRIPTION
**Pull request by Mohammed-JS-Khan**
Added snackbar notifications for all of the professor's actions

### CODE REVIEW

File: public/css/quizzard.css [hunk 1]
```diff
@@ -3,3 +3,32 @@ body {
...
+    padding: 8px;
+    position: fixed;
+    z-index: 1100;
+    left: 50%;
+    top: 7px;
+}
+
+#snackbar.show {
+    visibility: visible;
+    animation: fadein 0.5s, fadeout 0.5s 2.5s;
```
Comment (larryyueli) [inline public/css/quizzard.css:23]: Does this animation show in all major browsers?

Comment (esteeroramy) [reply public/css/quizzard.css:23]: Good thinking, let's get this tested in IE Firefox Chrome and Safari

File: public/js/admin.js [hunk 2]
```diff
@@ -244,13 +252,13 @@ var submitUserForm = function() {
         data: user,
         success: function(data) {
             if (data == 'failure') {
-                $('#result').html('User could not be added');
+		dropSnack(colours.FAIL_RED, 'User could not be added');
```
Comment (esteeroramy) [inline public/js/admin.js]: Indentation seems off here and in all lines of the form `dropSnack....`

File: public/css/quizzard.css (hunk 1 above)
Comment (Mohammed-JS-Khan) [reply public/css/quizzard.css:23]: Firefox and Chrome are good.

File: public/js/admin.js [hunk 3]
```diff
@@ -276,10 +284,12 @@ var submitUploadForm = function() {
         contentType: false,
         success: function(data) {
             if (data == 'uploaded') {
-                $('#upload-result').html('File successfully uploaded');
+		dropSnack(colours.SUCCESS_GREEN, 'File successfully uploaded');
                 setTimeout(displayAccountsTable, 3000);
```
Comment (esteeroramy) [inline public/js/admin.js]: Let's get this tested before we merge this

File: public/js/admin.js [hunk 4]
```diff
@@ -481,3 +492,12 @@ var sortAccountsTable = function(type) {
         }
     });
 }
+
+/* This function slides down a snakbar */
+function dropSnack(colour, msg) {
+    var x = document.getElementById("snackbar");
```
Comment (esteeroramy) [inline public/js/admin.js:588]: can we change quotes here to be `'` rather than `"`. just to stay consistent

File: public/js/admin.js [hunk 5]
```diff
@@ -481,3 +492,12 @@ var sortAccountsTable = function(type) {
...
+    x.style.backgroundColor=colour;
+    x.innerHTML=msg;
+    x.className = "show";
```
Comment (esteeroramy) [inline public/js/admin.js:591]: can we change quotes here to be `'` rather than `"`. just to stay consistent

File: public/js/admin.js [hunk 6]
```diff
@@ -481,3 +492,12 @@ var sortAccountsTable = function(type) {
...
+    setTimeout(function(){ x.className = x.className.replace("show", ""); }, 3000);
```
Comment (esteeroramy) [inline public/js/admin.js:592]: can we change quotes here to be `'` rather than `"`. just to stay consistent

File: public/js/admin.js [hunk 7]
```diff
@@ -1,3 +1,10 @@
+/* This is the index for referencing colours for the snackbars */
+var colours = Object.freeze({
```
Comment (esteeroramy) [inline public/js/admin.js:2]: some documentations for this variable would be helpful

File: public/css/quizzard.css (hunk 1 above)
Comment (Mohammed-JS-Khan) [reply public/css/quizzard.css:23]: Could we get @mehtadarshan1 (or @larryyueli ) to fork my repo at some point and tests if it works on safari? And @ahmedqarmout2 or @esteeroramy you guys work on windows so can you test IE?

How to test:
1. Fork my repo. ( https://github.com/Mohammed-JS-Khan/quizzard )
2. Pull on to your laptop, make an admin account and log in using Safari or IE.
3. Go to student and create a valid student.
4. Confirm if a green bubble slides down from the top, hangs for a second, then slides back up.

I'd do this myself but I don't have Windows or OSX and I can't find IE or Safari on the Arch User Repo.

File: public/css/quizzard.css [hunk 8]
```diff
@@ -3,3 +3,32 @@ body {
   padding-bottom: 20px;
   background-color: #eee;
 }
+
+#snackbar {
```
Comment (ahmedqarmout2) [inline public/css/quizzard.css:7]: Can we test this on smaller devices to make sure the fixed size will work

File: public/css/quizzard.css (hunk 1 above)
Comment (esteeroramy) [reply public/css/quizzard.css:23]: I think we should also be testing how this will look on a phone/tablet

File: public/js/admin.js (hunk 2 above)
Comment (ahmedqarmout2) [inline public/js/admin.js]: Maybe make this dropSuccessSnack(message)
dropErrorSnack(message) etc.
Right now the way you know if the snack represent a success or failure is by its color.

**Review by ahmedqarmout2 — CHANGES_REQUESTED** [review]
Nice work. Look at the comments and let's decide on what's easier and best to build on in the future :)

File: public/js/admin.js (hunk 2 above)
Comment (Mohammed-JS-Khan) [reply public/js/admin.js]: We'll they'd just be repeated blocks of code, the only thing that differs is the colour and the text, both of which are parameters of the dropSnack function. If we ever want to add more colours to represent something else we can just add them to the list at the top and call it a day.

Unless I'm missing some logic here X'(

**Comment by larryyueli** [comment]
Please add an entry for snackbar in your package.json, so that other people can do "npm install" to install all necessary package. I tried to tested but nothing showed up probably because I don't have snack bar installed.

**Review by larryyueli — CHANGES_REQUESTED** [review]
Please add an entry for snackbar in your package.json, so that other people can do "npm install" to install all necessary package. I tried to tested but nothing showed up probably because I don't have snack bar installed.

**Comment by Mohammed-JS-Khan** [comment]
@larryyueli i did not add any new lib or node module. Snackbar is made up of webkit stuff which is built into current HTML and CSS.

File: public/css/quizzard.css (hunk 8 above)
Comment (Mohammed-JS-Khan) [reply public/css/quizzard.css:7]: Not sure how but this is a good idea. Can alway be adjusted later when we go back for nicer styling.

**Review by ahmedqarmout2 — APPROVED** [review]

**Review by Mohamed-A-Mohamed — APPROVED** [review]
Looks good to me, any indentation/style we can fix after everything has been pulled.

**Review by esteeroramy — CHANGES_REQUESTED** [review]

File: public/css/quizzard.css (hunk 1 above)
Comment (mehtadarshan1) [reply public/css/quizzard.css:23]: Worked on IE and Safari it looks good

**Review by mehtadarshan1 — APPROVED** [review]

**Pull request merged**